mktrayicon.c

```diff
@@ -256,10 +300,99 @@ gpointer watch_fifo(gpointer argv)
 #endif
 			break;
 		case 'm': /* menu */
-			fprintf(stderr, "Menu command not yet implemented\n");
-			if (param != NULL) {
-				free(param);
+			if (onmenu != NULL){
+				//destroy the previous menu
+				for (int i = 0; i < menusize; i++){
+					free(onmenu[i].name);
+					free(onmenu[i].action);
+					onmenu[i].name = NULL;
+					onmenu[i].action = NULL;
+				}
+				free(onmenu);
+				onmenu = NULL;
+				gtk_widget_destroy(menu);
+				menu = NULL;
 			}
+			
+			menusize = 0;
+
+			if (param != NULL && *param == '\0') {
+#ifdef DEBUG
+				printf("Removing onmenu handler\n");
+#endif
+				break;
+			}
+
+			// This block makes sure that the parameter after 'm' is well defined
+			int commas = 0;
+			int bars = 0;
+			if (len < 3){
+				break;
+			}
+			if (param[0] == ',' || param[0] == '|'){
+				printf("Action string cannot begin with ',' or '|'. Use '\\' to escape\n");
```

Comment:
Is there any particular reason to explicitly disallow this?

Reply:
The first character isn't anyway checked in the `while` loop below but for consistency I disallowed the `delimiters` to be used there. If you want to use them it's better to `escape` them. Makes the logic for the whole parameter the same.